platforms/arm.go

```diff
@@ -0,0 +1,69 @@
+package platforms
+
+import (
+	"bufio"
+	"log"
+	"os"
+	"runtime"
+	"strings"
+)
+
+// Structure to identify ARM platforms
+type PlatformArm struct {
+	// Present the ARM instruction set architecture, eg: v7, v8
+	Variant string
+	// Present the features that could not be detectable through HWCAPS
+	Features []string
+}
+
+func isArmArch(arch string) bool {
+	switch arch {
+	case "arm", "armbe", "arm64", "arm64be":
+		return true
+	}
+	return false
+}
+
+func isPattern(str string, pattern string) bool {
+	// The pattern in /proc/cpuinfo ends with \t e.g Features\t
+	return strings.EqualFold(strings.Trim(str, "\t"), pattern)
+}
+
+// For Linux, the kernel has already detected the ABI, ISA and Features.
```

Comment:
Is this format standardized? Looking at a random box, this doesn't seem to be correct.

Reply:
Thanks for your reminder. It seems x86 and ARM has different space characters.
I will trim all leading and tailing space characters of the string.